### PR TITLE
Monorepo restructure

### PR DESCRIPTION
**Pull request by mhuynh5757**
**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

Moved the app into its own subfolder and created a server folder for the backend component.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,66 +1,29 @@
-# Twilio Voice React Native App
+# Twilio Voice React Native Reference
 
-This is a React Native App that demonstrates [Twilio Voice React Native SDK](https://github.com/twilio/twilio-voice-react-native) usage.
+## What is this Project?
 
-## Prerequisites
+The Twilio Voice React Native Reference App is an example implementation of the Twilio Voice React Native SDK and serves to inspire developers who want to leverage the power of Twilio Programmable Voice in their React Native applications.
```
Comment (charliesantos) [inline README.md:5]: >...implementation of the Twilio Voice React Native SDK...

Please add a link to Twilio Voice React Native SDK

File: README.md [hunk 2]
```diff
@@ -1,66 +1,29 @@
...
+Please follow the official React Native guides to set up your environment, and additionally the Twilio Programmable Voice documentation. TODO LINKS
 
-## Running the app
+### Structure
 
-Switch to the example directory
-```sh
-cd example
-```
+This repository is a monorepo hosting two major components.
```
Comment (charliesantos) [inline README.md:15]: Please remove this statement. No need to mention, we just show it.

File: README.md [hunk 3]
```diff
@@ -1,66 +1,29 @@
...
+  ```
+  app/
+  ```
+  The example application uses the Twilio Voice React Native SDK and features a dialpad to make outgoing calls, and registration to receive incoming calls.
 
-On a different terminal window, run on Android:
-```sh
-yarn run android
-```
+* A complimentary backend server under:
```
Comment (charliesantos) [inline README.md:23]: No need to mention complimentary

File: README.md [hunk 4]
```diff
@@ -1,66 +1,29 @@
...
 
-Start metro bundler:
-```sh
-yarn start
-```
+* A React Native application under:
+  ```
+  app/
+  ```
+  The example application uses the Twilio Voice React Native SDK and features a dialpad to make outgoing calls, and registration to receive incoming calls.
```
Comment (charliesantos) [inline README.md:21]: We should list the features (in a bulleted format) in the app readme instead.

**Review by charliesantos — APPROVED** [review]
+1. Just left a few minor readme comments.

File: README.md (hunk 3 above)
Comment (mhuynh5757) [reply README.md:23]: Ack, I was thinking of the mathematical definition of `complement` but it could be weird if it's construed the other way.

**Pull request merged**